pkg/storaged/crypto-keyslots.jsx

```diff
@@ -288,21 +296,40 @@
     }
 
     progress(cockpit.format(_("Checking for $0 package"), package_name), null);
-    return check_missing_packages([package_name], null)
-            .then(data => {
-                progress(null, null);
-                if (data.missing_names.length + data.unavailable_names.length > 0)
-                    steps.push({
-                        title: cockpit.format(_("The $0 package must be installed."), package_name),
-                        func: progress => {
-                            if (data.remove_names.length > 0)
-                                return Promise.reject(cockpit.format(_("Installing $0 would remove $1."), name, data.remove_names[0]));
-                            else if (data.unavailable_names.length > 0)
-                                return Promise.reject(cockpit.format(_("The $0 package is not available from any repository."), name));
-                            else
-                                return install_missing_packages(data, status_callback(progress));
-                        }
-                    });
+    return check_installed_package(package_name)
+            .then(installed => {
+                if (!installed) {
+                    return check_missing_packages([package_name], null)
+                            .then(data => {
+                                progress(null, null);
+                                if (data.missing_names.length + data.unavailable_names.length > 0)
+                                    steps.push({
+                                        title: cockpit.format(_("The $0 package must be installed."), package_name),
+                                        func: progress => {
+                                            if (data.remove_names.length > 0)
+                                                return Promise.reject(cockpit.format(_("Installing $0 would remove $1."), name, data.remove_names[0]));
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18698-20230425-113621-0ab0dcb3-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)